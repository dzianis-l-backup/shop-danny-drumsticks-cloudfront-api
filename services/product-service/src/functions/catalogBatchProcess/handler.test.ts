// import { SQSRecord, Context } from "aws-lambda"
// import { catalogBatchProcess } from "./handler"

// import { HttpStatuses } from "src/types"

// jest.mock()

// describe("product-service", () => {
//     describe("catalogBatchProcess", () => {
//         let topicArn: string

//         beforeAll(() => {
//             topicArn = process.env.SNS_TOPIC_CREATE_BATCH_PROCESS_ARN

//             process.env.SNS_TOPIC_CREATE_BATCH_PROCESS_ARN =
//                 "arn:aws:sns:eu-west-1:test:shop-danny-products-sns"
//         })

//         beforeEach(() => {
//             snsMock.on(PublishCommand).resolves({ MessageId: "chiky pooky" })
//             dynamoMock.on(TransactWriteCommand).resolves({})
//         })

//         afterEach(() => {
//             dynamoMock.reset()
//             snsMock.reset()
//         })

//         afterAll(() => {
//             process.env.SNS_TOPIC_CREATE_BATCH_PROCESS_ARN = topicArn
//         })

//         it("should process batch requests", async () => {
//             const result = await catalogBatchProcess(
//                 {
//                     Records: [
//                         {
//                             body: '{"title":"vicfirth5a","description":"Vic Firth 5A American Classic Hickory","price":1,"count":1}',
//                         } as SQSRecord,
//                         {
//                             body: '{"title":"promark5b","description":"The 5B is a standard diameter drumstick for the heavy hitter","price":1,"count":1}',
//                         } as SQSRecord,
//                     ],
//                 },
//                 {} as Context
//             )

//             expect(result).toEqual({
//                 statusCode: HttpStatuses.CREATED,
//             })
//         })
//     })
// })
