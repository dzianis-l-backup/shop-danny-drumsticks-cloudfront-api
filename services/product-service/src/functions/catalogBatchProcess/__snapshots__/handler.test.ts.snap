// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product-service catalogBatchProcess should process batch requests 1`] = `
{
  "payload": [
    {
      "payload": [
        {
          "description": "Vic Firth 5A American Classic Hickory",
          "id": "534f1263-bd2d-4f89-aa93-0ad9d24442c6",
          "price": 1,
          "title": "vicfirth5a",
        },
        {
          "count": 1,
          "product_id": "534f1263-bd2d-4f89-aa93-0ad9d24442c6",
        },
      ],
      "statusCode": 201,
    },
    {
      "payload": [
        {
          "description": "The 5B is a standard diameter drumstick for the heavy hitter",
          "id": "f2e8f5c7-bcfc-4cc3-ad0d-24160814c790",
          "price": 1,
          "title": "promark5b",
        },
        {
          "count": 1,
          "product_id": "f2e8f5c7-bcfc-4cc3-ad0d-24160814c790",
        },
      ],
      "statusCode": 201,
    },
  ],
  "statusCode": 201,
}
`;
